src/Symfony/Component/EventDispatcher/InvokableEventSubscriberTrait.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\EventDispatcher;
+
+/**
+ * Helper trait for single-event subscription.
+ *
+ * The Event name will be inferred from the FQCN of the event parameter.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+trait InvokableEventSubscriberTrait
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents(): array
+    {
+        if (!$eventName = static::$listenerEvent ?? null) {
```

Comment:
I can't use this shortcut because this fatal error on PHP 7.1.3 & 7.1.4 https://3v4l.org/NbQYC#v700, though that was fixed in 7.1.5 :(

Reply:
could we live with that or should we work around it ? https://bugs.php.net/bug.php?id=74188

Reply:
The workaround suggested in the bug ticket looks acceptable to me.